cmake/FindGMock.cmake

```diff
@@ -0,0 +1,105 @@
+# Finds the Googletest Mocking library
+
+# This file can be configured with the following CMake variables:
+# - gmock_ROOT_DIR
+# - gmock_INCLUDE_DIR
+# - gmock_LIBRARY
+# - gmock_USE_DEBUG_BUILD
+
+# Or, alternatively, the following environment variables:
+# - gmock_INSTALL_DIR
+# - LIBRARY_PATH
```

Comment:
Same question here as the one in `FindGLFW.cmake`. What does this mean for building the engine?

Reply:
Lets users specify alternate installation locations, if they couldn't be found the "normal" way.